### ISSUE TITLE
No way to change the "bits" unit = We'll stop recommending Bread Wallet as prefered iOS wallet.

### ISSUE THREAD

**Issue by Transisto** [voisine/breadwallet#299]
At Sastoshi Counter, Located at Bitcoin Embassy Montreal,  we have, over the last year and a half, been recommending Bread Wallet to all our iOS customers, in-store and online. We made wallet guides explaining how to use Bread Wallet, we printed and gave away more than 1000 of those.

It is one thing to explain to our customer that bitcoin has a changing price, that there are many prices, that there is no USD or CAD bitcoins, that there is an exchange rate between USD and CAD. Very basic maths that everyone involved in bitcoin know but that our average underbanked iPhone using customers doesn't.  It is just too much for us to have to explain that when they are asked to pay 0.03465 BTC they need to type 34650 bits.

I can more easily explain to our customers what are milli-bitcoin, but Bits are just to far remote a concept than even I can't count the zeros I need to add or remove to enter an amount.

The "Bit" is not an SI unit of bitcoin that make any technical sense, it's just something that has been made up by members of the community.

We find that not allowing people to use the bitcoin unit at this time is stubbornness and arrogance.

Other problems : 
We have scratched our head for a few days to figure-out where the balance was.  Tapping the  Bread wallet logo is absolutely not intuitive !!! And since 80% of the UI for sending coin is empty space there is just no reason not have a "see balance" button.

The Left/right scrolling is not intuitive either, I don't know of any other app that use this kind of behavior when they are not even space constrained.

Bread wallet is too minimalistic and in the wrong way. 

"Everything Should Be Made as Simple as Possible, But Not Simpler"   - Albert Einstein

**Comment by voisine** [comment]
@Transisto, first let me just say that we love frank feedback! Keep it coming. It's the best way we have to make the product better.

You should absolutely only recommend what you think is the best product available to your users. We always strive to provide the best product we can. This principle we stick to uncompromisingly. I'm really sorry that it has come across as stubborn arrogance. Maybe you only see it that way because you disagree on the best way to serve users.

Let me ask you a few questions. Your answers to these should help each of us understand the other's point of view.

Do you think that full bitcoins will be a usable denomination in the future when bitcoin becomes a major global currency? 

Obviously we feel that it will not be, so given that, is it better for everyone to switch to a more usable denomination today, or wait for the number of bitcoin users to grow first and then switch?

Do you think today that dealing with an mount like .03465 is easier or 34650? If the latter, wouldn't it make more sense to encourage the person making a request denominated in bitcoins to switch to more user friendly bits?

It's important for the whole community to switch together to the new standard, which is why we don't offer the option to change it. We recognize that it's inconvenient when many others haven't switched, which only demonstrates how important it is to have a single standard used by everyone. It will only get more difficult as the community grows if we don't switch now.

I believe you are in error about bits not being a SI unit. They are equivalent to micro-bitcoins. We use the unicode lowercase stroke b character to represent this in the app, and "bits" seems to be the term that has the most traction in the community, but if people start calling them "micro-bitcoins", we're more than happy to do the same.

other issues:
We decided not to display the balance when the app is first launched for privacy reasons. You don't want everyone who scans your address to see your wallet balance. Simply tap the lock symbol to reveal it.

The left/right swiping indicated by the dots is the most familiar UI paradigm there is for every iPhone user, since this is what users are presented with on the home screen every time they use their phone. We've received much praise for the minimalistic two screen interface. Just send and receive. Also by defaulting to an uncluttered send screen, the experience is optimized for a quick checkout experience.

Eagerly awaiting your responses! Please change our minds!

**Comment by voisine** [comment]
"The reasonable man adapts himself to the world; the unreasonable one persists in trying to adapt the world to himself. Therefore all progress depends on the unreasonable man."
-George Bernard Shaw

**Comment by dabura667** [comment]
> they are asked to pay 0.03465 BTC

Anyone telling someone an amount by voice is asking for trouble. You should show how every single Bitcoin wallet allows you to generate a QR code with the amount in the request. removing the need for anyone to worry about typing large decimal points or large numbers.

I have not manually input a bitcoin amount outside of testing wallet software for over 2 years.

**Comment by FrancisPouliot** [comment]
No joke this JUST happened again a few minutes ago, someone complaining he doesn't understand the decimals in Bread Wallet. This is so typical of Bitcoin: "experts" who have absolutely no idea of how the average and "real" user behaves.

Everywhere breadwallet users use Bitcoin they see 0.001 bitcoin and on breadwallet they see bits and are super confused. 

We sell bitcoins to around 20 people in person each day and countless more online. For all the iphone users, every single time, we have to spend about a minute explaining to them that no, they don't have thousands of bitcoins in their wallet.

Screenshot of the text I just got today (in french)
![image](https://cloud.githubusercontent.com/assets/7997682/10871211/8e89faa4-80ae-11e5-8e43-8d0088ffda07.png)

**Comment by FrancisPouliot** [comment]
Why would you not just have the option to switch to bitcoins and not bits? It's such an irrational decision. Also, it would be ideal to see the balance in the "send" page.

**Comment by Transisto** [comment]
> Anyone telling someone an amount by voice is asking for trouble. You should show how every single Bitcoin wallet allows you to generate a QR code with the amount in the request. removing the need for  anyone to worry about typing large decimal points or large numbers.

> I have not manually input a bitcoin amount outside of testing wallet software for over 2 years.

It has nothing to do with voice or not.  Not that I condone them but Electrum have not even bothered to implement URI support yet, so it's fair to estimate that most people using Electrum are still typing the the amounts by hand and don't mind much. (https://github.com/spesmilo/electrum/issues/1518)

If you force hide people's balance behind a password and an obscure button you should not also allow to freely see the trade history.  FYI this is currently how most people manage to figure out their balance while almost none of them care that their balance can be exposed. (I would thus make this hiding balance an optional feature.)

As for the Global currency thing, bitcoin is not there yet, and trust me **people will choose** to use other units on their own when it become convenient to them. (many altcoin trader count and talk in satoshis)

If today's average transaction size is 60$ or 0.1875 BTC that's 4 digit while 187500 bits is 6 digit and a new word to learn that already has a widely accepted meaning. 

"A bit (short for binary digit) is the smallest unit of data in a computer. A bit has a single binary value, either 0 or 1. "

**Comment by voisine** [comment]
@FrancisPouliot, I completely agree with you that when people don't all use the same the denomination, the experience is confusing and painful. Real users in the real world are struggling as a result, just as you say!

If you agree that the bitcoin denomination will be unusable in a future where Bitcoin is a major global curency, (I argue it's already pretty unusable today) Then you will understand why it's so important for us all of us to switch together to bits now rather than later. It will only get more painful the longer we wait!

When you encountered this difficulty, you had two options, you could ask us to switch to the bitcoin denomination, or you could ask the person making the payment request to use bits. Which of these options do you think will have the better outcome in the end?

If we give users multiple options to chose from, that only works against our goal of arriving at a single standard for the community, which as you rightly pointed out, is the most important thing.

**Comment by voisine** [comment]
@Transito, why don't we together go and lobby electrum to do the right thing. I will even code it up and submit a pull request.

We don't show transaction history while the wallet is locked, only transactions that don't have 6 or more confirmations.

I'm sympathetic to the idea that people will choose the denomination most convenient. That's a great argument. I'm only worried that this is not an individual choice. It's quite difficult to choose somethig different than what everyone else is using (this discussion is evedence of that!) this is my counter argument, that each users choice affects the whole community and impacts bitcoin adoption. I want to push for the right standard now while there is a chance.

Also the most widely accepted definition of "bit" is certainly not the smallest unit of data. That is a very recent definition that only computer nerds are even aware of. The word "bit" is a small amout, or what you put in the end of a drill, or in a horses mouth. Bit was also a unit of currency equal to 1/8 of a dollar long before it had anything to do with computer science.

**Comment by FrancisPouliot** [comment]
If I understand correctly, your purpose is not to improve the user experience but rather to impose to encourage "bits" as the standard. Actually, I was part of the standards committe of the Bitcoin Foundation and I strongly advocated for BTC - mBTC - uBTC. The general consensus was that "bits" should not be the standard.

We are very selective about the products and services we recommend. I like Bread Wallet's simplicity, but seriously people need to have the option to change to BTC if they want to. This will for sure improve user experience. Sorry to break it to you, but Bread Wallet is used primarily by "noobs". People who are a bit more expert will use wallets which have more functionality.

We had previously been recommending blockchain.info but have stopped doing so since they do not support BIP32 and encourage address re-use. I will never discourage the use of Bread Wallet, but I'm just letting you know that this tiny tiny lack of feature will cause us to recommend Airbitz instead. I estimate that around 1000 have installed Bread Wallet as a result of our recommendation (particularly since our bread wallet guide is featured on www.backpage.com)

![image](https://cloud.githubusercontent.com/assets/7997682/10871798/72d3b000-80bf-11e5-81f5-c9be008d58d7.png)

**Comment by voisine** [comment]
@FrancisPouliot, our purpose is absolutely to improve the user experience first and foremost. I'm a bit shocked this wasn't clear to you already! Sometimes in order to achieve something better, we have to make sacrifices short term. Take for example when a social media company changes their interface. This causes short term pain as users are subjected to something new and unfamiliar, which they will inevitably complain about bitterly, but if the right choice is right, it will benefit those users and new users to come in the future.

As mentioned before, I'm happy to call them "micro-bitcoins", "uBTC" or anything else. We use ƀ as the currency symbol to denote them in the app and only mention the name "bits" in one tip bubble. We will eagerly change it to whatever the most popular name is. 

Thanks so much for you work with our friends over at the bitcoin foundation on a uBTC standard denomination. You can tell how important we think it is. And thank you for the compliment - "noobs" are exactly the people we are targeting! :) I hope every bitcoin business will do the same. We need noobs, and lots of them to make bitcoin a success.

I would hope that you choose not to recommend web wallets, not because of bip32 or address reuse, both very important financial privacy issues, but because web wallets run on insecure platforms and subject users to unacceptable risk of malware theft, phishing and man-in-middle attacks. We will never release a product that we believe puts users funds at risk.

Rather than trying to get wallets like breadwallet to support the btc denomination, would you be open to working together to get other wallets and bitcoin services to switch to uBTC as the default?

**Comment by dabura667** [comment]
Perhaps it's because I live in a country whose currency is much closer to parity with bits than it is BTC, but almost everyone I've introduced it to "gets" 10000 of something being like the price of a small snack. I could understand where people who think 0.75 of something is a small snack might be surprised when trying to understand numbers such as 100000.

I think that bits is what every wallet should be using, and we should all forget about the "bitcoin" unit.

In fact, I would argue that you should set up users on Android and Windows Phone with Copay or some other wallet and immediately tell them to switch their setting to bits, and just start explaining with bits, and keep everyone on the same page.

If bitcoin is gold, I would equate 1 whole bitcoin with a giant gold bar. Talking about how I will pay you with 0.01% of a gold bar doesn't make sense, which is why a lot of governments created gold coins of varying purities and sizes and weights and named them different things for different countries.

cutting the bitcoin into smaller chunks and renaming it something actually makes a lot of sense to me...

Sure, it won't make sense to someone who is trying to sell someone who's never seen gold before an old 20 crown gold coin when they've got all these Gold-bug traders pushing them to refer to it as "0.0001% of a gold bar... get the unit right, or else you are a noob"

**Comment by moveson** [comment]
I'm new to BTC and chose breadwallet because roughly 100% of reddit people recommended it as far and away the best wallet for iOS. I couldn't be more pleased with the product. As a long-time iPhone user, I found the two-screen swipe interface completely intuitive.

Voisine's argument (particularly if you believe BTC will increase in value by orders of magnitude) has to be correct. 

Keep up the great work!

**Comment by neuhaus** [comment]
I think "bits" is not self explanatory, it's confusing.
Ideally the app should use BTC or mBTC (switchable in settings).

**Comment by Transisto** [comment]
Bit is not self explanatory at all, we're talking about an app running on a
digital device using "bit" for meaning something else than unit of
information.

Are you asking us to lobby other wallet so they remove the ability to
change the units ?

**Comment by neuhaus** [comment]
Well, to be fair, the term "bits" is explained the first time you launch the app and whenever you tap on it.

**Comment by voisine** [comment]
@Transisto, baby steps. I think the first step would be just to get every wallet to support micro-bitcoins as the default (I don't care what they are called) Is that something we could work together on?

I'll also just point out again that we don't actually use the term "bits" anywhere except one tip bubble. We use "ƀ" to represent micro-bitcoins. I'm happy to change the one instance of the term "bits" to whatever the most widely accepted term is for micro-bitcoins. Your opposition to the term however I think might be colored by your work with computers. Neuro-typical humans excel at deriving meaning from context, something that computers are famously inept at. This is actually what makes them so frustrating for the uninitiated. A great example of this is how when you hear the term "bits" you automatically think of information science, because that's the context they are being discussed in, yet you know full well that "bits" is much more commonly used in the english language to mean "little pieces" of literally anything.

**Comment by zotrik** [comment]
I've been using bread for about a week now and I'm changing that, simply because of the developper's stubbornness which doesn't allow the user to personalize the app to their convenience. I like the BTC unit better than the bits unit. Trying to force a new standard to the community definitely is stubbornness.

**Comment by dabura667** [comment]
> I've been using bread for about a week now and I'm changing that, simply because of the developper's stubbornness which doesn't allow the user to personalize the app to their convenience.

Gmail doesn't allow you to choose which encoding you send your SMTP messages in... so I am switching to... uhh... oh wait, no one does that anymore...?

Someone's gotta start in the movement to abstracting away artifacts of the old, confusing ways.

Lots of my peers complained when the first GUI based e-mail clients came out and "OMG HOW CAN I SEND MAIL TO MY FRIENDS WHEN THEIR CLI SOFTWARE ONLY ACCEPTS BASE64 AND YOUR SITE DOESN'T LET ME CHOOSE OMG!" (note: OMG was not a thing at the time)

You'll grow out of it...

But until then, have fun with whatever app you choose to switch to. Just don't get too attached to the app if you'd rather be attached to the BTC unit, as apps will start dropping it like flies.

**Comment by keo** [comment]
I absolutely agree with @voisine. Let's not use BTC as the main unit. This is a problem of geeky early adopters who've seen more than 100 BTC in one shot.

Think of emerging markets. Think of all the newcomers. They won't get their head around 0.0023 units of something.

It's time to start telling people to switch to uBTC/bits. That's what I tell everyone right after showing them how to backup.

**Comment by neuhaus** [comment]
Am 08.11.15 um 17:33 schrieb Barnabas Debreczeni:
> Think of emerging markets. Think of all the newcomers. They won't get
> their head around 0.0023 units of something.
> 
> It's time to start telling people to switch to uBTC/bits. That's what I
> tell everyone right after showing them how to backup.

I agree that BTC isn't ideal.
However, "bits" is bad. It's non-standard, it already is well-known with
a different meaning (yet still IT related) and it isn't
self-explanatory. µBTC is much better in that regard.

**Comment by keo** [comment]
> However, "bits" is bad. It's non-standard, it already is well-known with
a different meaning (yet still IT related) and it isn't
self-explanatory. µBTC is much better in that regard.


Tell any US person to use µBTC. It isn't even a printed sign on a keyboard.

Last year was about this debate, and I agree with @voisine that breadwallet should switch to whatever naming gets more traction. Currently it is bits, even if we don't like it, or it is not scientifically perfect. But my Mom understands it.

And people in UK are perfectly fine with having "pounds" for two completely different things – and it works, because the context always sets the meaning.

So, "send me 10,000 bits" obviously means you're talking about money. "this file is huge, one gigabyte" - you obviously talk about data size. "My connection is 20 megabits on my phone" - you're obviously talking about the speed. People aren't that stupid.

**Comment by voisine** [comment]
@zotrik I understand that it's awkward and frustrating to switch to something new that you're not used to. Nobody likes change. I would encourage you though to not overlook the other advantages that breadwallet gives you, such as being the only SPV bitcoin client that runs on a hardware encrypted, malware hardened platform. When making a BIP70 payment, the verification of payee identity also happens directly on the malware hardened device. With typical hardware wallets, you have a trust a potentially infected host machine for payment addresses.

Also remember that if you want your bitcoin holdings to become more valuable over time, we will all have to make bitcoin simple and easy to use for new comers from all backgrounds. If we're all using different denominations, or requiring people to understand arbitrary precision decimal numbers, we will hurt bitcoin adoption, and hurt our own pocketbooks in the process.

**Comment by voisine** [comment]
@zotrik I will always stubbornly stick to what I think is best, but I very often change my mind when presented with well argued counter points. I'm honestly open to being convinced I'm wrong. I just haven't heard any arguments that the cost (which I acknowledge) outweighs the benefit in this case.

**Comment by mkz899** [comment]
yes,it's named "Bitcoin", of course use "bit" as natural unit for daily usage

**Comment by Logicwax** [comment]
what's the big deal about putting in a switch buried in settings, for those who want BTC?  Obviously people want it.   I'm sure it wouldn't take much time to implement.

Ideally, it would be nice to have the first three digits after the decimal be scaled in different size, and the three following a smaller size (and raised up), and the three following...........Square's "Cash" app does this with USD dollars and cents.     Tinting a different color or high-light would add to the visual effect as well.   This would help as a visual aide in leading people into thinking about bits and uBits and so-on.

**Comment by voisine** [comment]
@Logicwax, it's important for the community to pick one standard. The pressure you feel to use full-bitcoins is a great example of this. It's what you're used to, it's what most others are still using, it's a pain doing it differently than everyone else. This is why it's so important that we all switch *now*, as it will only continue to become more difficult the longer we all wait. This is why we don't have an option buried in settings. If we did, people might use it, and that would confuse new users, hurt bitcoin adoption, and make the product worse.

**Comment by neuhaus** [comment]
@voisine The argument goes both ways. Noone that I know of is using "bits" except for this client. Everyone else is using mBTC and uBTC if they need units that are between Satoshi and BTC.

People (especially those that use metric units) are familiar with the SI prefixes, they use them every day with distances, weights, speeds, etc.

**Comment by jprichardson** [comment]
>  Noone that I know of is using "bits" except for this client. 

Ummm, my wallet.... 
![Exodus bits](http://www.exodus.io/startup/common-files/img/content/exodus-wallet@2x.png).

> Everyone else is using mBTC and uBTC if they need units that are between Satoshi and BTC.

I wish the United States went SI, but they didn't. So this complicates things. I see the argument in favor of using SI units, but I'm with @voisine on choosing a user friendly unit that can appeal to all - and I think 'bits' is a fine choice.

**Comment by voisine** [comment]
"bits" are uBTC. We only call them bits in one tip bubble, which I'm happy to change to uBTC if that's the name that becomes popular.

We can't keep BTC as the standard denomination since it's already pretty useless for typical transactions, and bitcoin is still tiny. We are designing to make bitcoin a major global currency.

**Comment by keo** [comment]
Americans not familiar with SI can use 1/16ths of a BTC if that's what make them happy  :)

Let's hope BTC mainstream adoption will force them to learn SI.

**Comment by voisine** [comment]
1/16th can be exactly represented with IEEE 754 double. It would solve floating point rounding errors! :)

**Comment by Transisto** [comment]
Bread Wallet devs want to impose their standard onto people, they see users as a heard that can't be allowed to make the wrong decision. 

Typical "big government know best" ideology that is so common in the whole Apple ecosystem. That's why, when asked, we recommend people don't use iPhone to start off instead we suggest devices that are open, flexible and compatible with, for example, hardware wallets via either USB or NFC.

I agree that "mBTC" and "bits" make more sense in the long term but guess what, **they are compatible between each others**. Stopping people from using the base unit "Bitcoin" by principle and hoping they forget it ever existed is stupid, especially when you don't have significant market share and can barely keep-up with features.

It's the early adopters who will suggest new users to use Bread Wallet or not. If we were to suggest Electrum, my guess is ~95% of users would stick with the default unit mBTC because it's easy for them to make the connection between "0.2541 BTC" and "2541 mBTC" (but not 254100bits!). 
The switching cost of fiddling in the settings to change the default is worth it for uBTC. (see Default effect, https://en.wikipedia.org/wiki/Default_effect_(psychology) )

To finish up this rant,  Bread is too minimalist / cryptic and not intuitive.
As I said earlier, you're not the one who spend countless hours explaining to people how they have to multiply and divide everything by 1 000 000, how to see their balance on their wallet, to show them to swipe left and right to go to send/receive, to tap their bitcoin address to show the menu to copy/paste.

![image](https://cloud.githubusercontent.com/assets/388716/11854932/092d9f02-a415-11e5-9e2c-8f0aca988cc0.png)

Share, Send, Receive and Activity buttons ... look straightforward to me.

**Comment by FrancisPouliot** [comment]
@Transisto @voisine agreed 100% - copay is much better and they are a lot more concious of their userbase. Bread had potential, but in the end it's just a flop.

It's a shame breadwallet devs couldn't listen to their biggest promoters. I just realised when you google "breadwallet" Satoshi Counter is 6th and 7th result.

Too bad!

**Comment by voisine** [comment]
@Transisto, I'm really encouraged that you see that "bits" make more sense long term. This is a point of agreement that I feel we can build on. Have you given any additional thought to my suggestion that we work together to convince other wallets to support uBTC as the default denomination? That would solve the problem of users having to do conversion in their head in a way that promotes bitcoin adoption both now and for the long term. I'd love to get your thoughts on that.

We chose to launch first on iOS due to the security profile of the device, being both hardware encrypted by default and having app sandboxing and enforced code signatures to harden against malware. I hope that you only recommend holding private key material in secure devices. Very few people, even very technical people, have a strong enough understanding of security to safely manage digital cash on a platform that isn't secure by default.

Your point about tapping the receive address to copy it being too cryptic is well taken. We felt like the tip bubbles explaining it when tapping the QR would help, but having a share icon I think would make it clear at a glance. People mostly refuse to read.

@FrancisPouliot we always listen to our users, but they don't all agree with each other, and sometimes we have to make hard choices. More options and settings is often not the right way to improve a product when considering the needs of all users.

**Comment by Logicwax** [comment]
* This is why we don't have an option buried in settings. If we did, people might use it, and that would confuse new users, hurt bitcoin adoption, and make the product more cluttered with strange options many users don't understand.*

If I use it, how does that confuse other users?    Aren't new users the one's not likely to go deep into settings and fiddle?   Default settings is where you can preach to the masses, but I'm making an inquiry for the power users.


Anyway, I'd like to see more attention to my UI idea.  I think by using color-shading, and font-size scaling (like how Square's CASH app does) is a great way to passively embed your idea into others heads.   It's really nice to have UI queues that allow people to see instantly sets of three digits and UI queues that demonstrate digit significance.  This will help people move to this vision you are trying to push people into.

**Comment by voisine** [comment]
> If I use it, how does that confuse other users?

Because then you haven't adopted the community standard. You won't ask for merchants and services you transact with to use the standard. You won't be using the standard that any new people will be using who you happen to transact with.

I don't actually think that I'm trying to push anything on anyone, I'm just trying to make the best product I can. I'm convinced long term that means having one bitcoin denomination that isn't full bitcoins.

When Facebook changed the the homepage to the timeline it was big change that lots of people complained bitterly about, but they felt it would make the service better once people got used to it. In the short term however, it definitely made the product immediately worse for everyone who was presented with something unfamiliar instead of what they were already comfortable with. Similarly, Apple also chose not to support flash on iOS devices because they felt it would make the product worse in the long term, despite the short term problem of having lots of flash sites being inaccessible.

**Comment by Logicwax** [comment]
Well, people are going to use their local currency denomination for quote a while.  

With that said, what about that UI idea?  I'm saying there's a graceful way to help people out.

**Comment by voisine** [comment]
@Logicwax I like how the square cash app handles cents, and we could do something similar for the satoshis portion of an amount you enter, but pretty much no one enters values all the way down to the satoshi just yet.

Currently the app uses full number formatting with commas separating each grouping, so the number preceding an initial comma is mBTC, and the number preceding a second comma is BTC. Changing the font size or color to further delineate might be nice. We'd have to experiment and see how it feels.

**Comment by brutesque** [comment]
This wallet is seriously nice, but the bits stuff is totally confusing.

**Comment by voisine** [comment]
@brutesque it is certainly confusing when most people aren't using bits yet, but they are actually a lot less confusing for anyone not already used to full bitcoins (nearly every person in the world) especially once the standard catches on.

There are other currencies in popular use today where a typical beverage might cost 5,000 units, but none a where it costs 0.005 units

**Comment by brutesque** [comment]
I have seen a lot of very long explanations for leaving out a simple option to change the default. The end result is that it is mainly confusing with all the other sites and apps, because they don't agree. Every payment I do with the app get's me confused.

**Comment by voisine** [comment]
@brutesque how about a short one then... the solution is to get those other sites and apps to change their default to bits.

This solves the confusion, makes bitcoin more usable to the rest of the world, and ensures we never have to switch again. As difficult as this solution is to implement, it will continue to get worse if we wait.

**Comment by brutesque** [comment]
I disagree that the future of bitcoin will stand or fall because the the bits/btc issue. I think this is a non-issue. People have to come here to have long discussions with you, to find out you have a personal opinion you are forcing user to comply with.

It's my favorite bitcoin wallet, but it's truly annoying every time to not understand how many bitcoins i have. For all purposes, I don't care about the bits. Never have, and only in the distant future will they matter.. maybe.

**Comment by samuraisam** [comment]
> It's my favorite bitcoin wallet, but it's truly annoying every time to not understand how many bitcoins i have. 

However many million bits you have is how many full bitcoins you have.

**Comment by brutesque** [comment]
Yeah, so people from zimbabwe won't be confused :)

**Comment by brutesque** [comment]
I honestly think, this is all just personal opinion. I don't share the whole bits ideology at all. I think all the discussions here proof that not everyone agrees. Why would you let it stand in the way of a small feature, that many people will embrace. You can totally leave it on bits by default.

**Comment by brutesque** [comment]
We are still pricing bitcoins in BTC on every exchange.

**Comment by voisine** [comment]
> I disagree that the future of bitcoin will stand or fall because the the bits/btc issue.

I also disagree that the future of bitcoin will stand or fall because the the bits/btc issue.

> you are forcing user to comply

I honestly don't feel like we're forcing anything on anyone, just providing the best product we can with a eye for long term. It's open source.

> only in the distant future will they matter.. maybe.

It was only two years ago that a bitcoin was $10. Give it a second!

> people from zimbabwe won't be confused

The vietnamese dong is worth 0.1 bits, indonesian rupiah 0.16, columbian peso 0.73, south korean won 1.98, and several others are under two bits. A not-insiginifcant portion of the world's population already uses a denomination in this value range. Exactly none use anything like full bitcoins.

> discussions here proof that not everyone agrees

Not everyone likes every standard, but we use them because there is far more value in using what everyone else uses than going on our own. This is why people still use full bitcoins despite all their problems. The switch won't happen unless we encourage it. We're encouraging it by making (hopefully) great bitcoin software that people want to use, despite it only being offered in bits.

> We are still pricing bitcoins in BTC on every exchange.

As @samuraisam pointed out, the conversion is quite simple... the number preceding the second comma is the number of bitcoins. If you can manage arbitrary precision decimals in your head, I bet you'll have no trouble.

**Comment by Transisto** [comment]
In a world where everything is paid in gold people will use what's most convenient ; be it ounces, grams or milligrams. Would you be going out of your way to force people into thinking the one unit is the micro gram because you believe the price of gold will be worth 100 000$ an ounce at some point in the future.  

There is one similarity between all the example currency you gave. They all fucked up their money supply.

You basically want every sheeptards stuck in Apple land to be thinking like there is no such thing as a bitcoin. Beside the annoyance this will be creating confusion now and in the future.

**Comment by voisine** [comment]
@Transisto, if I was making an app for using gold as a transactional currency, I wouldn't put in an option to use an obsolete, mostly useless unit like ounces, even if all the big forex exchanges still quoted rates in ounces.

> going out of your way to force people

I don't that word means what you think it means. All I can do is provide the best product I know how. I can't force anyone to do anything.

> They all fucked up their money supply.

Indeed, this has happened to many currencies. It's happened enough that large portions of the world population are used to using denominations in this value range. Anyone who uses money can handle it. There are however large portions of the world's population who will never be able to handle arbitrary precision decimals, as using full bitcoins would require.

> sheeptards stuck in Apple land

You mean people who care about privacy and security?

**Comment by Transisto** [comment]
"Going out of you way" meaning spending more time debating not adding a feature than it would take to add it.

> There are however large portions of the world's population who will never be able to handle arbitrary precision decimals

Nice, so these people can either keep the default value to bits or mBTC OR can use full precision up to the last Satoshis if they want.

> You mean people who care about privacy and security?
No I mean the other 99.999% that use apple devices.

**Comment by voisine** [comment]
> "Going out of you way" meaning spending more time debating not adding a feature than it would take to add it.

It's not of question of difficulty, it's a question of what the right thing to do is. I've made my argument that the right course of action is to lobby other wallets and bitcoin software to start using uBTC as the default. I haven't yet heard any arguments in favor of adding a full BTC option other than, "it's what most others are still using". So, lets address that concern and start encouraging others to switch. @Transisto, can I count on your help?

**Comment by pinheadmz** [comment]
Chiming in -- Breadwallet, which I love and use regularly, is the only Bitcoin-related thing in my entire world that uses bits. No other wallets, merchants, exchanges, block explorers or other software I interface with uses bits.

My main problem is there's no reference point when you look at something like "60 bits," which was the fee on my most recent Breadwallet transaction:
"6,000 Satoshis" makes sense to me because I know there are no more numbers hidden to the right.
"0.00006 Bitcoins" makes sense to me because I know there are no more numbers to the left.

**I would be most in favor of a compromise in which all "bits" notation in the app was displayed XX.YY so at least it looks like Satoshis: "network fee: 60.00 bits"** Especially if your goal, @voisine is to make Bitcoin look like other global currencies (formatted XX.YY)

In addition, in my mind right now I know 1 Bitcoin is around $450, so when I'm shopping:
0.1 BTC is what a nice shirt would cost
0.01 BTC for a coffee
0.0001 BTC for a ChangeTip tip
so what's "50,000 bits"?

**Comment by brutesque** [comment]
@voisine your opinion is that this is the right thing to do. But a lot of users struggle with it. The length of this page shows you how much trouble they have with it. Your stubbornness is getting silly. And you are not right in your argument by fact. You're just right by opinion. A opinion that is actually very rare. Who are you to know what is best for bitcoin. The only reason i'm not changing the code myself and build it for iphone is because I don't have a developer account. But as soon as someone will create a wallet with the same features, I'll be gone. I've got a feeling I won't be alone.

**Comment by brutesque** [comment]
I have 62 euros on the wallet. It shows b174.871. That is in no way better than, 0.17 btc.
The fiat amount only uses 2 digits. Tell me how is expressing that with 6 digits better?!?

**Comment by brutesque** [comment]
And who are you to tell me I have to adjust to how you want to see bitcoin. Huge discussions with your userbase about your superior opinion is just silly.

**Comment by keo** [comment]
@brutesque you're holding it the wrong way. :)

Jokes aside, I'm with @voisine on this one.

When Bitcoin will $5000 or $50,000 it will be increasingly difficult to keep track when denominated in BTC, and this issue will come back again and again.

10,000 is easier to read & adjust than 0.01. I live in Hungary, and we use big numbers on a daily basis.

My monthly internet bill is 8,000 HUF. Three months worth of gas is 80,000 HUF. Way easier to tell difference compared to 0.08 and 0.008. This is how our brains work.

If you're in the US, then yes, you're used to small numbers. But going from small numbers to big ones (and then back, when the Bitcoin price skyrockets next time) is way easier than going from 5 to 0.0005.

For daily usage of a currency, 2 decimals at most is what most people can get their heads around, no matter how you fight against that - this is a FACT.


BTW, if you have OSX El Capitan, you can fork the code, modify it, compile it, and put it on your iPhone for free.

**Comment by dabura667** [comment]
> In addition, in my mind right now I know 1 Bitcoin is around $450, so when I'm shopping:
0.1 BTC is what a nice shirt would cost
0.01 BTC for a coffee
0.0001 BTC for a ChangeTip tip
so what's "50,000 bits"?

5 coffees.

I really don't see what's so hard about this. Are people being facetious here or do we really not know how to multiply large whole numbers mostly consisting of zeroes in our heads?

**Comment by nshopik** [comment]
> But going from small numbers to big ones (and then back, when the Bitcoin price skyrockets next time) is way easier than going from 5 to 0.0005.

Man I hate counting these zeros on invoices to make sure they got it right.

**Comment by keo** [comment]
you have to do it anyway, either on this side or on the other side of the decimal.

On the left side you usually group them with thousand separators. That also helps a lot with seeing the ballpark figure.

Nothing does and no one does that for zeroes on the right side (for decimals).

**Comment by nshopik** [comment]
Not really, since prices still in fiat you end up in not so round number

**Comment by dabura667** [comment]
Exercise!

A. 0.003628

OR

B. 3628

A: *looks at number* hmm, is that 0.0003628? no, wait it's 0.003628, ok.

B: *looks at number* hmm, 4 digits. that means thousands, so about 3-4 thousand bits. k.

No one would look at 3628 and see 36280 but someone looking at 0.003628 could definitely accidentally read it as 0.0003628 or 0.03628

**Comment by keo** [comment]
One more exercise:

89,245

OR 

0.00089245

So is it 0.0089245 or 0.000089245 or 0.00089245? Can you spot the difference at first look?


Now compare to:

8,924  vs.  89,245  vs.  892,450     -- you can easily spot the ballpark figure here.

**Comment by voisine** [comment]
@brutesque, I get it. The way breadwallet does it is strange, you're already used to something different, and right now uBTC is arguably not as good.

Try to understand though that our goal is not to serve the existing bitcoin community even though we hope they will like what we've built. Our goal is to help make bitcoin into a major global currency used by billions of people. Using full btc, or multiple denomination standards will hurt that goal.

**Comment by pinheadmz** [comment]
@voisine What do you think about adding the decimals in all displayed amounts?

before:  60 ƀ
after: 60.00 ƀ

1) It looks more like all global currencies, ie $60.00 etc.

2) The fact that it looks like "6,000 satoshis" makes it way easier to understand how this value relates to 1.00000000 BTC

**Comment by brutesque** [comment]
@voisine Oh please. The goal should be to make a wallet that is usefull right now. Don't force people into using what you think the everyone should use in the future. I love bitcoin, but I do not agree everyone needs to move to uBTC. And I do not pretend to know what people will or should use in the future. I do now which denomination I need to use right now.

This is the last I'll say, because I don't want to spend any more time on this discussion. Clearly this is never going to change.

**Comment by brutesque** [comment]
@keo Thanks for the tip. I didn't know it could be done for free.

**Comment by FrancisPouliot** [comment]
@voisine correct me if I'm wrong, but the change is rather trivial so at can't we just remove "BTC" once the price of Bitcoin hits 100k$?

**Comment by voisine** [comment]
@brutesque, I don't know what to tell you, I really can't relate at all to ignoring bitcoin's potential (nothing less than becoming the new monetary standard for planet earth) in favor of focusing on what it is today, which is barely more than a toy. That just doesn't interest me.

@pinheadmz, that's not how people in places with large denomination currencies do it, and I disagree that it's easier to relate to BTC. With uBTC, you can just look at the number preceding the second comma. ƀ1,000,000 = 1 bitcoin

@FrancisPouliot the change is definitely trivial, the code is flexible and supports any denomination. It's really just a configuration option. The problem with waiting to switch is that it will get more and more difficult as the number of bitcoin users grows, and more people become used to full bitcoins. Just look at how much grief we're getting already, then imagine multiplying by 500x when the price if 500x.

**Comment by pinheadmz** [comment]
@voisine I hear you, but there's not always a second comma:

1,000 ƀ

I look at that and I either have to imagine: 

1,000[00 satoshis]

which I understand is:

[0.00]1,000[00 BTC]

That number "1,000 ƀ" is like floating in undefined space without a left edge or a right edge. As the number in bits gets bigger the number of preceding zeroes before 1 BTC changes (10,000 ƀ or 100,000 ƀ). So whenever I see a bits value it's a two-step process to visualize the 8-decimals that create 1 BTC, which is my reference point.

**Comment by voisine** [comment]
> visualize the 8-decimals

We love you @pinheadmz, but neuro-typicals don't do it that way. ;-)

**Comment by pinheadmz** [comment]
Fair nuff! <3

**Comment by Transisto** [comment]
What other initiative have you done to ease adoption of a standardized unit ?  There are tens of thousands service and wallets of which 99.9% don't use Bits.  I was thinking of funding a crowd sourcing effor to fill a wiki table to sort out what default unit they each company is using by default and what option do they give to their users.

I'm all for the bits but If you don't allow advanced user to use the unit of their choice, even if it mean doing gymnastic in the settings, Bread Wallet risk not being recommended as much.

**Comment by aviadoffer** [comment]
Kindly add the BTC option. You can leave Bits as default but most people are used to BTC.
I like this wallet a lot but right now the Bits display is too confusing to me and I am not using the wallet just because of that. Thanks for listening.

**Comment by voisine** [comment]
@aviadoffer you can help us move the Bitcoin community to the bits standard. Bits are only confusing because you got used to btc first. Since btc is mostly useless for typical transactions it's important we make the switch, so Bitcoin can become a successful global currency used by millions. Please ask the vendors of other Bitcoin software you use to default to bits, so people like yourself don't end up confused. Using full bitcoins just isn't an option for global mass adoption.

**Comment by brutesque** [comment]
Are an expert on what will bring us global mass adoption? Don't you you get that it's only your opinion?

**Comment by aviadoffer** [comment]
I don't disagree. but change takes time. After all we still didn't change
the 1MB block size although it's clear it's crippling us.
All I'm asking is to add it as an option, most people (the ones not used to
BTC or new to Bitcoin) will use the default and Bits will become more
popular.
In the future world where 1 BTC will be 10X what it is today Bits will sure
rule ... but change takes time. Let's not push people away from the wallet
because of it.. or worse let's not cause fork here aka Classic breadwallet
... hate to see that mess.
Totally respect what you and the team created, just asking to make it
friendly to us all.

**Comment by aviadoffer** [comment]
One more comment and I will shut-up. You and the team are the talented folks who created this wallet and the reason we are all passionate about it is because we appreciate it and see it's outstanding value compare to what's out there. All we are asking is to give us a small voice and not push us away.
While you guys are the experts we are the end users and asking for small changes to allow us to adopt it.
This is open source and we thank for that, but let's make it "open" by listening to others. I don't feel I'm the only one asking for this, hence this huge long thread. All that for a pretty small change.
Now I will shutup. 
How about a vote? 
Thanks again.

**Comment by voisine** [comment]
@brutesque, yes I am an expert on what will bring global mass adoption for bitcoin. (I've designed and built several products that now have millions of users) That doesn't make me right, but it does give me confidence to follow my convictions. It is my expert opinion that switching to bits is the best course of action to help achieve this goal.

@aviadoffer, I value your opinion and don't want you to "shut up". Please continue to share it. I'm sorry if people feel that I haven't been listening. I really am trying to listen and consider with an open mind. So far the arguments I've heard are that bits are confusing if you're already used to btc, but to me this seems like an argument in favor of switching now since it will only become more difficult. Although the change is small, in fact it's just a configuration change, it's still an important one. Having multiple options I also think is a bad idea. This will slow down the switch by perpetuating the existing btc standard, confuse new users and complicate the product. The value in a standard is that it's shared by everyone. The more people who use it, the easier it is for further people to use it. I agree this is inconvenient for those who switch first, but sometimes we have to do something difficult in the short term for a larger long term benefit.

I do hope someone forks breadwallet and adds lots of "power user" features for bitcoin geeks (like me!) that we don't have the engineering resources to justify due to the limited audience. It would be cool and relieve some pressure on this project to reduce it's focus on maximum security and simplicity for non-experts.

**Comment by aviadoffer** [comment]
I think people think you are not listening because almost everyone who starts using breadwallet is looking for the option to change the display and it's not there. They all end up opening a ticket/request and wasting your time. like you said, instead of putting all your efforts on the security aspect which is clearly your forte you spend time answering the same question over and over.
I felt silly when I opened a ticket realizing (when you answer after 0.03ms) that the same request was opened many more times before (clearly I didn't look prior to opening).
No one disagree that future is for Bits, but today with a wallet that has few thousands on USD/EUR BTC makes sense for most people.
I bet if I will look here in a month there will be hundred more of the same and you will keep answering them over and over.
The freedom to choose our way to pronounce (or display) should not be foreign to you, this the beauty of the English language.I say one thousand three hundred dollar you say thirteen hundreds dollar ... and in other languages they even say it in the other direction ... nothing wrong with that... this is not math where rules have to follow, this is language/display and should be freedom.

Default - Bits (cause that's the future)
Option - BTC (cause now it makes more sense than bits)

If people think this is the correct way please ACK.

**Comment by voisine** [comment]
@aviadoffer my argument against having multiple options is that it complicates the product with additional options that will be confusing for new users. It's already quite difficult just to get people to understand that bitcoin is a separate currency from USD or Euros. It's definitely the wrong choice long term, and by adding it short term we make the switch more difficult and take longer. 

The only people complaining are the relatively small community of existing bitcoin users. We still have the only popular wallet in the industry that connects directly to the bitcoin and runs on a secure-by-default platform, so even with the annoyance of having to use bits, those existing bitcoin users are still choosing breadwallet. I'm convinced this is the right way to do it, even though we get complaints from those who are inconvenienced in the short term. Understand that we take that inconvenience seriously and would never subject users to it if we didn't think the long term benefit was worth the very real short term cost.

**Comment by aviadoffer** [comment]
OK. Thanks for your time. I'm moving on. It's Copay for me and I will not recommend breadwallet (not because of the bits/BTC, because open project should be open and allow the voice of the users to impact the dev cycle. with someone like you this is not going to happen based on this small sample).
Good luck and thanks for making this wonderful wallet. You are a talented man. Maybe in few years you will learn it's not enough in life.

**Comment by voisine** [comment]
@aviadoffer I'm sorry we weren't able to reach agreement. I believe I have listened with an open mind and made well reasoned responses to your concerns. Know that the voice of users has had a large impact on the product, even though we can't accept ideas we feel moves the project in the wrong direction.

**Comment by Transisto** [comment]
Could you respond to 

> "What other initiative have you done to ease adoption of a standardized unit ? There are tens of thousands service and wallets of which 99.9% don't use Bits. I was thinking of funding a crowd sourcing effor to fill a wiki table to sort out what default unit they each company is using by default and what option do they give to their users.
>  I'm all for the bits but If you don't allow advanced user to use the unit of their choice, even if it mean doing gymnastic in the settings, Bread Wallet risk not being recommended as much."

**Comment by voisine** [comment]
Thanks @Transisto, sorry for not responding earlier. So far I've spent all my bitcoin related efforts building breadwallet, but I think the time may be close for some additional outreach to get others to join us. I really like your idea of a wiki showing which projects support bits, which default to it, and which are bits only.

As of now the big focus is on the blocksize limit, as it should be. Maybe a good time to launch would be after seg-wit is deployed. What do you think?

@aviadoffer, you seem like a very reasonable person, so it does bother me that you feel your thoughts haven't been listened to (to the point you'd avoid recommending the project to people) That's something I want to improve on. Can you help me understand what I could do differently? I felt like our discussion was productive and we were communicating well.

**Comment by pinheadmz** [comment]
Whenever I am teaching a friend about bitcoin for the first time, I always make them download breadwallet to their phone, because it is easy and SPV and wonderful, then I send them some Bitcoin, and show them how to shop online with it. But at every step I need to instruct them about the difference between bitcoin and bits. And why the price on that bitpay invoice doesn't look like the amount of money their wallet is sending. Or the fact that I didn't just send them 3,000 Bitcoins.

 "Welcome to Bitcoin! We have two units right now and you need to learn them both."

**Comment by voisine** [comment]
@pinheadmz yes, I agree this is quite unacceptable. Since btc won't work long term, we should focus our efforts on getting bitpay to support bits as the default.

**Comment by aviadoffer** [comment]
> Aaron, first thank you very much for putting this project together. Second, and I think more importantly you raised the bar on how security should be done with our wallets. 
Once you put the project as an open source while  the credit is yours now the project becomes part of the community. Meaning  that it's up to the people to decide how it moves forward. Clearly the governance should be done by the experts and specifically here all decisions related to security and implementation should be by those experts. However other enhancements typically related to the visibility and usability of the project should be left to the people the end users that are using it every day. As an end-user I feel you're not willing to give control on any aspect of the project or listen to the majority of the people. We haven't proven majority on the specific issue of bits. But I feel that even if we would prove it you would still not change something as simple as that. The project and this wallet could do very well but your crippling it by forcing your opinions on usability matters. I think you're brilliant and this project is fantastic however you catch more bees with honey and I fear that you're lacking that understanding. 
Great leaders are not just smart but they know to pick their battles and they know to compromise to satisfy the crowed and lead. 
Again thanks for your time and ear.
I think in few long years you will understand me better. I was once the same when I lead projects. It's part of us being young and never wrong.  It's all good.

**Comment by jprichardson** [comment]
> Once you put the project as an open source while  the credit is yours now the project becomes part of the community. Meaning  that it's up to the people to decide how it moves forward. 

The project does not become a part of anything outside what the creator chooses (unless someone forks - which @voisine has explicitly stated he'd wish for someone to fork for power users). As an open source developer, there are zero obligations to the community. Now to build a successful open-source project, that requires a delicate balance of leadership, vision, and listening to the community. Vision and usability matter. I think that's @voisine perspective.

> However other enhancements typically related to the visibility and usability of the project should be left to the people the end users that are using it every day. As an end-user I feel you're not willing to give control on any aspect of the project or listen to the majority of the people.

It'd be a horrible idea to delegate usability to end users. I'd bet most of the beautiful products in existence were not designed by committee, rather, were designed by the vision of one or two individuals.

> Great leaders are not just smart but they know to pick their battles and they know to compromise to satisfy the crowed and lead.

Great leaders set a vision, recruit a tribe with shared belief, and along with that tribe, execute upon that vision. What you're describing is a politician.

> I think in few long years you will understand me better. I was once the same when I lead projects. It's part of us being young and never wrong.  

I really don't think that this is case of never being wrong or lacking any sort of wisdom. I think this is case of @voisine having a vision for the future of Bitcoin and orienting his product in accordance with that vision. If that means that people will be unhappy in the meantime, then so be it.

Please know that my previous statements were out of sincerity and gentleness and no hostility was intended :) We all want Bitcoin to succeed and I think it's really cool that different people are building products in different ways to make this happen.

**Comment by voisine** [comment]
@aviadoffer thanks for your thoughts. I really do appreciate it!

In this case, this distribution of the code at least is owned and controlled by the breadwallet investors. They've entrusted me to move it forward, but I will only hold that position as long as they think I'm the best person to do it. For this reason I have to do what I think is right. Others are also free to use the code in other projects with different goals (several do and it's great!)

Im not in a position where I'm free to delegate decisions to the majority, but I can and do listen and try my best to understand their position. I hope I can do that well enough that you feel comfortable continuing to recommend breadwallet.

In this case I think the disagreement is simply that you have a short term personal goal (the best experience as simply and quickly as possible right now!) that is in conflict with this projects long term goal of helping mass bitcoin adoption. If I was only considering your goal, I'd make the change right now. Part of leading is both to be willing to change your mind when you are wrong (I have done on several issues with breadwallet), but also to stick by your convictions in the face of even strong opposition. 

Yes we must pick our battles, but this project is growing quickly in popularity, recently moved ahead of blockchain.info in the App Store most popular rankings, and I'm convinced we can actually win this one. If we eventually become the most popular wallet, especially among non-technical new bitcoin users, other services will feel a lot of pressure to switch their default.

**Comment by keo** [comment]
Let's not make breadwallet something like this :) (although this drawing is exaggerated, it illustrates the point quite well)

![apple_technology_techguygadgets_1](https://cloud.githubusercontent.com/assets/101425/12556514/8288c65c-c386-11e5-8d95-ce26914fad09.jpg)

**Comment by pinheadmz** [comment]
> owned and controlled by the breadwallet investors. 

I didn't know that... is there a profit model? How are they expecting return on a free open-source app?

**Comment by aviadoffer** [comment]
@Transisto  and @voisine  thanks both.
I step down. I wish the project good luck. I hoped there would be a simple setting to allow end user BTC/Bits but I failed to communicate the need.
Thanks again for brining this awesome project to life.

**Comment by voisine** [comment]
@aviadoffer I think you communicated the need very well. It's just that there are competing concerns, and sometime it has to come down to a judgement call.

@pinheadmz the model we're working on will be similar to that of free open-source web browsers. Our goal is to get as many users as possible, and then have one or two optional built-in features that increase the value of the product to users, but that can also be monitized, kind of like the browser search bar. The simplest one will be the ability to purchase bitcoin through a broker/exchange, based on your jurisdiction and preferred legacy payment method. This improves the value of the product for new users who don't know how to get bitcoin.

**Comment by TheButterZone** [comment]
Add me to the "denominate in full BTC or bust" list.

**Comment by ffxsam** [comment]
Geez, why is there such a huge discussion about this? It's ridiculous, and I say that as someone with years of experience in development & UX.

Just put an option in the settings to allow the user to choose between µBTC, mBTC, and BTC, and everyone will be happy. It's silly to want to impose something as miniscule as µBTC on other apps like Electrum. Some people always deal in very large transactions, and seeing a number like 8572102 µBTC is not practical. mBTC is a nice middle ground, if you really had to pick just one.

When in doubt, allow people the option to choose what works for them. It's a no-brainer.

**Comment by ffxsam** [comment]
Of course, on the flip side, this is open source, so the authors are not obligated to do anything. Anyone is welcome to fork this project, which might be a good idea. I love its simplicity and trustworthiness, but I struggle with some of the UI/UX decisions, like having my balance in such a tiny font at the top of the screen.

**Comment by voisine** [comment]
We have a UI redesign in progress. The feedback about the font is good. I was a bit aggressive on making it thin for aesthetics. With the new San Francisco fonts in iOS 9, you can fine tune the font thickness more precisely.

@ffxsam, you have to remember that a bitcoin was worth only $15 just a few years ago. We don't want to have to go through this fiasco of changing the standard unit again.

mBTC has 6 decimal places which is still confusing for a large portion of the world's population, doesn't work with huge amounts of existing financial/accounting software, and will eventually need to be changed anyway.

µBTC won't need to be changed again even when bitcoin replaces all existing currencies. It has two decimals like everything else, and although it's a bit unwieldy right now, it's already a larger unit than many currencies used every day by hundreds of millions of people, including the local currencies of Columbia, Vietnam, and Indonesia (and the South Korean Won is worth less than 2bits)

**Comment by voisine** [comment]
@ffxsam, regarding wrangling other bitcoin apps and software into support the change, I know it seems like a daunting task from the outside, but we're quite well plugged into the bitcoin developer and business communities. When the time is right to start a campaign pushing it, I think we can make a lot progress on this front.

**Comment by Transisto** [comment]
https://imgur.com/PDCpVy2

**Comment by ffxsam** [comment]
@voisine That's a good point about the South Korean Won, and other currencies where one of their dollars equals 1 µBTC, or in the case of the KRW, 0.5 µBTC!

Still though, I don't see anything wrong with the addition of an option so people can choose the denomination that works best for them and their currency, or the transaction sizes they're used to.

**Comment by Transisto** [comment]
I don't like that the setting are below the list of transactions.

I don't use Bread but I see people looking for how to select USD instead of CAD but they can't find it.
Then I have to tell them their wallet is using a specific exchange to fetch the price.  Which ? No way to tell or to change it.

Sometime I wonder if Bread is in the bitcoin wallet business or into experimental UI design.

**Comment by ffxsam** [comment]
Some critical but honest feedback: if there are investors involved, please hire a UX engineer. Other wallets have much better UX & UI (Blockchain, Coinbase), but BreadWallet wins out over those as it's the only one that:

1. Lets you maintain your own private keys.
2. Works offline.
3. Doesn't use a hosted wallet of any kind.

Hence, IMO, it's the most secure.

**Comment by voisine** [comment]
@Transisto in the about screen it mentions rate info is from bitpay.com/rates, though I agree this is not easy to notice. The idea there is that users will want to see the exchange rate they are most likely to encounter when making a purchase. bitpay is the largest bitcoin payment processor and uses multiple exchanges to find the highest price available for a market sell order.

We decided to limit the app to just two screens (send/receive) since this is what people want to do quickly when launching the app. Viewing your history is then just one additional tap. Settings is rarely needed or used, so being slightly inconveniently located is a good trade-off to keep the rest of the app uncluttered.

@ffxsam I appreciate the critical feedback, it's the only way we will get better. As I mentioned we are working on a some new UI design ideas. We've heard the opposite about the UI/UX design (from professional designers), especially relative to other bitcoin apps. Did you perhaps first learn to use bitcoin on one of these other apps and so something different feels unfamiliar?

**Comment by mkz899** [comment]
it would be nice keep current version and will not update if the new UI version release , it's perfect for the guy who agree with the simplify UI design and the "bit" unit design .

Voisine, Thanks for the great work again.


-Mike

**Comment by ffxsam** [comment]
@voisine No, I actually started my Bitcoin wallet experience with BreadWallet, then tried the others. My impression is strictly from comparing and analyzing the interfaces.

I'd recommend a few things that would greatly improve the experience of using BW:

* Make the balance more prominent (bigger font).
* Don't make people have to tap the lock icon to see their balance and transactions. I'd do one of two things: #1, have no passcode/TouchID protection and rely on the user to have their phone properly protected/locked; #2, have passcode/TouchID to open the app, and then the user can access all aspects of the application.
* The swipe-between-pages experience feels strange. I'd just use an iOS UIToolbar at the bottom, with: Transactions, Receive, Send.
* Have a main menu that has the other options such as settings, import private key, rescan blockchain, etc.

If you wanted to make the app super friendly to newbies, you could even do a way with the Bitcoin terminology. E.g. instead of "import private key," make it "transfer paper wallet." Instead of "rescan blockchain," maybe "sync with ledgers." Just a couple ideas.

**Comment by ffxsam** [comment]
And I know you're resisting the option for the user to choose between BTC, mBTC, and µBTC.. ;) But I'd strongly recommend it. Some currencies work better with µBTC, but with others, it really depends on the sizes of our transactions. Sometimes BTC makes sense, other times mBTC is better. IMO, we developers don't dictate the conventions that users follow; users' needs dictate the application requirements. Meanwhile, you can talk to the community and push the agenda of universally using µBTC, and once it's agreed upon, *then* modify BreadWallet to be bit-based only.

Just my two cents.

**Comment by voisine** [comment]
Thanks @ffxsam that's great feedback.

I agree with 1)

As for 2) I can see that it's more common, but I'm concerned about the trade-offs. It means for instance that you have to double unlock (triple unlock if your phone was asleep) to make a payment. Once to unlock the phone, once to get into the app, and once again to authorize the payment.

3) This is something we're experimenting with. The initial reason I chose the swipe action with the two dots is because it's the most familiar UI paradigm for all iPhone users, mirroring the home screen. It's also quicker once you learn it, and it leaves all that beautiful negative space on the screen for a minimal look and feel. But we will be doing a/b testing to see if my theory holds up in the hands of real world first time users.

4) We essentially have this now, we just also put your tx history there. The idea there was that send/receive is the main purpose of the app, and reviewing your tx history is more for accounting, but this is also something that we're experimenting with to see if my theory holds up to real world use.

I agree that users dictate application requirements, and the needs of users of all kinds from all backgrounds and levels of technical expertise are what we're aiming to serve. In general more options and features make an app useful in more niches, but it comes at the expense of simplicity. This is something real world users crave, and something sorely lacking in the bitcoin landscape.

**Comment by ffxsam** [comment]
>As for 2) I can see that it's more common, but I'm concerned about the trade-offs. It means for instance that you have to double unlock (triple unlock if your phone was asleep) to make a payment. Once to unlock the phone, once to get into the app, and once again to authorize the payment.

This is assuming someone has their phone auto-lock immediately. Some people may have a 15-min delay before the phone locks, which is why they'd want the extra layer of security by having BreadWallet require TouchID. What if they step away from their desk and forget their phone, which is still unlocked? Someone like me who has it auto-lock immediately, I'd just have BreadWallet not have any passcode/TouchID enabled at all. Give the user the power to decide how they want to use the app. (And I think passcode/TouchID to authorize payment should always be there)

Totally agree about simplicity.. you don't want to overdo it and add every option possible and clutter up the app.

Good luck, Aaron! I'm looking forward to seeing what you do in future updates!

**Comment by dabura667** [comment]
Bitcoin's early adopters are a group of tinkerers, so obviously having a menu with 500 options to tweak and change is understandable. Nothing wrong with Mycelium.

breadwallet is fine as is. The only thing I'd change would be UP UP DOWN DOWN LEFT RIGHT LEFT RIGHT B A START to add your own node IP to the connections.

**Comment by abgit** [comment]
> it's named "Bitcoin", of course use "bit" as natural unit for daily usage

@mkz899 amazing :)

**Comment by peter-conalgo** [comment]
I've made a pull-request which adds a settings to switch between "bits" and "BTC". 

[More detail on pull request 387 here.](https://github.com/voisine/breadwallet/pull/387)

**Comment by xrishox** [comment]
i dunno if you care about this, but i've been becoming increasingly skeptical of mycelium's long term plan, so I've been looking into moving to another wallet. I've been aware of breadwallet for a while. I love that it's  real spv. I also really love that it has such a strong focus on security. I quickly downloaded it and sent a test transaction to it only to discover it was in bits. I looked through the settings and couldn't find a way to change it. bits just do not work for me. they do not compute correctly in my brain. it is really unfortunate because i was quite looking forward to using your app. i think you are being quite stubborn over something that is quite petty. i hope you reconsider.

**Comment by voisine** [comment]
@xrishox, thanks for taking an interest in breadwallet! Our goal is to help bitcoin become a major global currency, and full BTC as the standard unit just won't work if we are to succeed. I hope you can understand that in order to change the standard, it's going to require some short term discomfort, for you and others. The tiny fractions of a unit that most existing bitcoiners have gotten used to are only going to get way smaller still.

**Comment by Logicwax** [comment]
Regarding #2: every banking app / money app on my phone just requires one
touchID auth to get into it.     With an iphone7 I just use my home button
to wake and unlock the phone in one push.   If I go into circle or
Coinbase, it's holding my finger there while launching the app.    When
making a payment I hold it there again.  This is very easy UX.    Having to
tap the top bar of breadwallet and then hold my finger just to see the
balance seems silly and unnecessary.  There's no reason to cater to those
who don't properly lock their phone as that is such a narrow corner case.
Every user I see use breadwallet does this first thing when they launch the
app.   Why wouldn't they?   Why would you go into the app and not need to
know your balance?     Again, catering to narrow workflows.



And this bits / bitcoin / uBits domination arguments have to stop.    There
is no reason for the devs to not at least offer the option to change it in
settings.   This goes to show the devs have no desire to listen to their
users.

**Comment by voisine** [comment]
@Logicwax I really find it incredible that you would say we haven't been listening. Just look at the enormous time and energy put into this thread alone to explain the design decision and address everyone's objections. The option you want is even available in a code branch that's just a few clicks away. Is it really so hard to understand why we don't want to perpetuate a standard that will limit bitcoin adoption, and continue to become more and more unusable as use continues to grow? Really?

**Comment by Logicwax** [comment]
What's funny is you think that denomination CHOICE will limit bitcoin
adoption and usability.  DEFAULT configs are very underestimated in the
field of UX/UI (this is why highly technical people can't understand why
iMessage is so popular compared to the fragmented clusterf*ck that is
several android messaging apps).   Your voice, as a developer, is sustained
with the default config that you push onto your users.   Do not confuse
CHOICE with that.


As for "just a few clicks away" you mean I have to compile for my own
phone?    Game is already over at that point.   I'm a technical user and
can provide my own workarounds, but this UX complaint I have is for others
who are simply going on the AppStore and downloading a bitcoin wallet.
Provide an option in settings.   You seem to be really against providing
any options (even hidden ones if you want) in settings.  Why?   What for?
The amount of "enormous energy" you've put into arguing this could have
been spent on simply adding the choice in settings.


If you think that bitcoin adoption will be limited because of trivial
things like denomination, then no offense, but that's the myopic view that
I see very prevalent in the bitcoin community (and any other high tech
fields).   Bitcoin suffers a huge debt of people who understand UX/UI, who
understand how *normal* people actually use applications and computers.
If I had a dime for anytime I meet an extremely smart intelligent computer
scientist whose working on something bitcoin related that no one would
actually use, I'd be rich.  This is why it took google so long to finally
realize that all the PhD's in the world can't design decent UI/UX
of software and to consider actually putting money towards design.


Here you are professing what UX/UI you think would make or break bitcoin
adoption, yet your users are telling you otherwise.    I get that it makes
sense in YOUR head.   You're intelligent with technology.  You're really
smart and already are on the bleeding edge of something like a technology
such as bitcoin. Don't let that get to your head about how others will use
it.  I guarantee you that denomination will not be an issue in the adoption
of bitcoin.  There are WAYYYY more problems that geeks are overlooking that
are going to hold back bitcoin.    I sincerely hope you're joking when you
think this will be an issue......are you aware that the majority of users
will never care to interact with bitcoin at it's real value?    They won't.
  Like the TCP/IP stack did for the web and trading cute cat videos, users
will simply rely on software to represent bitcoin in their local currencies
or at least one real global currency that currently has a foothold, such as
USD or the euro or whatever.   Bitcoin has a huge ton of problems to work
out before it ever is even considered a retail POS payment system, if it at
all ever becomes one (I don't believe it will, other systems will solve
that problem and bitcoin will become a core settlement system / perhaps
breadwallet will use lightning in the future or something).   Spend your
time on other pursuits and listen to your users.    Arguing over
messageboards / email lists isn't the same thing as listening to your
users.

**Comment by ffxsam** [comment]
<img src="http://i.giphy.com/oaZk0WNSO7fXi.gif">

**Comment by voisine** [comment]
@Logicwax I agree with most of your points, bitcoin has bigger problems than the size of the standard unit, clearly.

Choice is generally good, but it also has a cost. Most obviously in product simplicity, but even more importantly when there is a lot of value for everyone to use same thing, i.e. standards. The bitcoin denomination is one I believe has importance for bitcoin adoption. It is a UX issue as you say.

Regarding local currencies, one of the most frequent support issues we see are from people who don't understand that bitcoin is a different currency from their local currency. "The dollar side going down but the bitcoin side is staying the same! What's going on? Where's my money?!"

**Comment by pinheadmz** [comment]
Every time this thread flares up I offer my personal favorite compromise: continue using bits, but just display the decimals. Like `130,345.00 bits`. That way old users who are used to eight decimal places can still see them and the developers' commitment to a more universal currency unit remains intact. Just look at my example. For me, an  older user who got used to "whole bitcoins", it's still very easy to interpret that as abiut 0.130 BTC 

If the price goes up another order of magnitude or more, we're all going to need to see the satoshis anyway.

**Comment by voisine** [comment]
@pinheadmz this is not a bad idea, but don't a lot of existing bitcoin UIs only show 6 decimal places, or even just the minimum precision? Also I think the people who complain the most are actually used to counting how many 0's there are between the decimal point and the most significant digit.

**Comment by dabura667** [comment]
How my mind got used to bits in the early stages:

1. imagine 3 digits after the thousands comma (fill to the left with zeroes)
2. imagine "0." in front of that
3. imagine the thousands comma disappears.

At first, it takes about 2 milliseconds every time I look at it, but eventually I got into the habit of just thinking of the price by dividing it by 10,000 (right now BTC is 70,000 yen ish) so if the price is 72,356 yen, I think "100 bits is 7.2 yen" and someone in Korea would probably think "10 bits is 7.2 won" and when we get to $1,000 Korea will reach parity with bits, $10,000 and JPY will reach parity with bits.

but tbh, breadwallet shows the value in my local currency anyways, and 90% of my usage is BIP70 links which have the amount to send set out for me anyways.

My stance is still with Aaron.

**Comment by FrancisPouliot** [comment]
It seems that the BTC option in Bread has been there for the last year.

**https://github.com/voisine/breadwallet/commit/c3c3f432903893d8715a803e4a303252d9ef14cb**

So you CAN actually switch to BTC. Even as a very experience bitcoin users interacting with all the wallets, I could not figure out on my own, so it's definitely not something that the average bread wallet user that's new to bitcoin and just needs to add credits to an online service balance can easily figure out on his own. But, the option IS there, so it seems that Bread Wallet is pro-choice after all!

Here is how to do it on Android:

1. Set up the wallet
2.  open menu
3.  go to settings
4.  click on local currency
5.  At top, you will see $1 = b 1 000 000 
6.  Swipe left over the conversion equation to get:
7. - $1 = mB 1 000
8. - $1 = B 1

![image](https://cloud.githubusercontent.com/assets/7997682/20201843/124f6174-a789-11e6-85ba-04e1885931a2.png)

**Comment by FrancisPouliot** [comment]
I would strongly suggest to inverse the conversion equation: everybody knows easily what the price of Bitcoin is, but not instinctively what 0.272 bitcoin is.

1 Btc = 700 rather than 1$ = 0.001428 ?

**Comment by voisine** [comment]
@FrancisPouliot listing how many bits there are in a dollar makes a lot more sense than a small fraction of a dollar per bit. Remember, we're trying to lead the way in changing the bitcoin community's standard denomination here, to grow bitcoin adoption and make it usable for everyone, even the half of the world with below average math skills.

**Comment by dabura667** [comment]
![img_4248](https://cloud.githubusercontent.com/assets/6586275/20204213/1e710916-a811-11e6-9070-99030a37c6a2.PNG)

Can confirm iPhone can also switch to BTC.

For the record, I promptly switched it back to bits. :-P

**Comment by Logicwax** [comment]
Thanks Francis! This is great. Now I can read my bitcoins like a regular
floating point number as that is how my brain processes things.

**Comment by abgit** [comment]
@FrancisPouliot, you are the man ! :)
The only problem here is that it's a buggy implementation because, after changing to btc, we cannot send 0,1 btc (but we can send 1,1 btc). Seems that decimals don't work if amount is smaller that 1.


@voisine
> we're trying to lead the way in changing the bitcoin community's standard denomination

keep dreaming

**Comment by dabura667** [comment]
@abgit no bug here. I just tested on iPhone and Android, both of them let me type in 0.001 BTC and sent it properly. (no display bugs or anything)

Perhaps you actually have it set to mBTC or still set to bits and you are trying to send below dust limit or something?

**Comment by abgit** [comment]
@dabura667, i tried again and get same issue.
On iPhone, after selecting BTC, if i start with a zero i can add "," but then it only accepts zeros..

When i try to setup amount 0,01 btc is displayed as  "0 B ,0" (comma and decimal is displayed after btc symbol)

**Comment by keo** [comment]
Your locale is probably using comma (" ,") as decimal placeholder.

This is a typical bug of payment apps that don't honor locale settings.

I don't know if this is the case here, apart from sounding a lot like it.

**Comment by voisine** [comment]
haven't had a chance to look at it yet, but we're not so careless as to forget about honoring locale settings :) We use NSNumberFormatter.decimalCurrencySeparator

My guess is bug with handling all the edge cases for locals that put the currency symbol after the number

**Comment by voisine** [comment]
@abgit we're sometimes ahead of blockchain.info in app store rankings. We've also had some inside looks at the active user stats for some of the largest players in the industry, and we're not as far off as you think.

**Comment by abgit** [comment]
Here goes a [movie](https://dl.dropboxusercontent.com/u/1839678/IMG_9609.MOV.mov) about the decimals problem on btc :)

**Comment by keo** [comment]
BTW iOS 10 already shows the new Bitcoin unicode symbol correctly...  (U+20BF) :)

**Comment by abgit** [comment]
right.. maybe there's some incompatibility of breadwallet-on-btc with ios 9.3.5?

As you can [see in this is movie IMG_9609.MOV](https://dl.dropboxusercontent.com/u/1839678/IMG_9609.MOV.mov), i get decimals after btc symbol :)

**Comment by abgit** [comment]
Anyone on this bug?

**Comment by dabura667** [comment]
It's hard to test for such an old iOS version.

Try changing your iPhone region to United States and your language to English, then ser if the problem still exists.

If it does, then probably an old iOS problem, if it doesn't then it may be a locale setting problem.

**Comment by abgit** [comment]
@dabura667, i have tested and this is not a ios bug. This is a locale bug.
After changing ios region and language to PT (Portuguese) i get this strange behavior. On EN region and language works like a charm.

**Comment by dabura667** [comment]
Also, have you tested whether it works if you type 0 , 0 0 1 instead of only , 0 0 1 ?

**Comment by abgit** [comment]
Both are buggy.

**Comment by gustavonalle** [comment]
Another suggestion: create a new unit called **octabit**, denoted as "⅛ƀ" which has a quite trivial meaning: an eight of the "soon to be ubiquitous" bit.

**Comment by i1o** [comment]
@voisine  Please accept my gratitude for developing the best in class SPV wallet and releasing it under open licence but accept as well my two millibitcoins in this discussion. This is of course quite a personal point of view, but since 2011 when I was first engaged into Bitcoin, I have never seen a place where people were dealing in terms of bits. I am personally sick of that denomination (am I an old dog who knows that bit is a binary digit and nothing else?) and I always have to process from one denomination to another with a discomfort. I find you decision regarding that denomination not well evaluated, but I don't care much about it, thanks MIT. My prediction is that the world will never use "Bits" in terms of bitcoin amount transferred. For the smallest transaction there are satoshis, for everything else there are mbtcs (and in rear cases btcs). That is because it is _thousand_ times easier for our brain to operate with values that are identical in their meaningful digits and easily compared to the parent value. Are you kidding me that understanding the equivalence of "0.03465"BTC and "34650"Bits is easier than the equivalence of "0.03465"BTC and "34.65"mBTC? To be clear, in my opinion your decision is neuro-mathematically wrong.

Thanks

**Comment by mkz899** [comment]
Unfortunately, you are not right, bits will be mainstream in future.

On 4 May 2017, at 1:56 AM, i1o <notifications@github.com<mailto:notifications@github.com>> wrote:


@voisine<https://github.com/voisine> Please accept my gratitude for developing the best in class SPV wallet and releasing it under open licence but accept as well my two millibitcoins in this discussion. This is of course quite a personal point of view, but since 2011 when I was first engaged into Bitcoin, I have never seen a place where people were dealing in terms of bits. I am personally sick of that denomination (am I an old dog who knows that bit is a binary digit and nothing else?) and I always have to process from one denomination to another with a discomfort. I find you decision regarding that denomination not well evaluated, but I don't care much about it, thanks MIT. My prediction is that the world will never use "Bits" in terms of bitcoin amount transferred. For the smallest transaction there are satoshis, for everything else there are mbtcs (and in rear cases btcs). That is because it is thousand times easier for our brain to operate with values that are identical in their meaningful digits and easily compared to the parent value. Are you kidding me that understanding the equivalence of "0.03465"BTC and "34650"Bits is easier than the equivalence of "0.03465"BTC and "34.65"mBTC? To be clear, in my opinion your decision is neuro-mathematically wrong.

Thanks

**Comment by kivodefly** [comment]
Not trying to argue but just my opinion. 

the whole discussion sounded like ppl are trying to decide whether Km, M,Cm or mm is going to be a main stream. Personally, i use bread wallet and i am ok with the way it is, however I have to say having the ability to switch between measuring units is better than locked in one. for new comers "bits" is easy to read, however for those who have quite a few BTCs will find them a hard time to read the gigantic numbers followed by 12 or more 'zeros' at the end.  :)

**Comment by keo** [comment]
> For new comers "bits" is easy to read, however for those who have quite a few BTCs will find them a hard time to read the gigantic numbers followed by 12 or more 'zeros' at the end. :)

I'd have never thought people have problems with looking at their wallet / account balance and saying: "hmm... too big numbers in here... I don't like having so much money" =)

(I am OK with seeing big numbers as my wallet balance :)

**Comment by kivodefly** [comment]
I know what you mean, i'm totally ok with big numbers  in the wallet. the more the better :P
 let me give you an example of what i meant:
let's say Bob has 1294.254 Bitcoins, in wallet A it displays 1294254000 bits, WalletB it's 1294.254 Bitcoins.

Now, one day Bob see the bitcoin value went up to $3000, he wants to see how much his bitcoins worth in total roughly.

my question is which number is easy to calculate (in your mind)? 1294254000 x 3000 / 1000000 
or 1294.254 x 3000 ?

The idea of using Bitcoin as a measuring unit is mainly for the ease of use in these kind of scenarios where you really want to quickly calculate a rough number in $.

cheers.
